src/main/java/io/github/cottonmc/cotton/gui/widget/WButton.java

```diff
@@ -22,10 +24,15 @@
 
 public class WButton extends WWidget {
 	private static final Identifier DARK_WIDGETS_LOCATION = new Identifier("libgui", "textures/widget/dark_widgets.png");
+	public static final Insets DEFAULT_ICON_INSETS = new Insets(2,2);
+	private static final int BUTTON_HEIGHT = 20;
 
-	private Text label;
+	@Nullable private Text label;
 	protected int color = WLabel.DEFAULT_TEXT_COLOR;
 	protected int darkmodeColor = WLabel.DEFAULT_TEXT_COLOR;
+
+	protected int iconSize = 16;
+	protected Insets iconInsets = DEFAULT_ICON_INSETS;
```

Comment:
Imo vertically centering the icon and offsetting it horizontally by 2px makes more sense as a default, so the insets system isn't really appropriate here.

(It also doesn't make sense since it has a "bottom" component which is meaningless in a constant-height button)

Reply:
Haha here you are totally right.
I used it here primarily to save some varibles.
I will rework it so that instead of Insets, it centers the icon vertically according to it's size - and instead of introducing any variable, i will offset the icon left and right by 2px.